### PR TITLE
GraphQL - Custom Max Node Limit Analysis

### PR DESCRIPTION
**Pull request by toyhammered**
# What

A clone of [github graphql limitations](https://docs.github.com/en/graphql/overview/resource-limitations) for node limits.

# Why

We don't want people to make super duper long queries that will do bad stuff to our system!

Sources:
1. [graphql-ruby analysis code](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/analysis/ast/query_complexity.rb)
2. [running queries on their own](https://graphql-ruby.org/api-doc/1.11.4/GraphQL/Analysis/AST.html#analyze_query-class_method)
3. [ast info](https://graphql-ruby.org/queries/ast_analysis)

# Checklist

- [ ] All files pass Rubocop
- [ ] Any complex logic is commented
- [ ] Any new systems have thorough documentation
- [ ] Any user-facing changes are behind a feature flag (or: explain why they can't be)

- [ ] All the tests pass
- [ ] Tests have been added to cover the new code

### CODE REVIEW

File: app/graphql/analysis/max_query_complexity.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def analyze?
+      true
+    end
+
+    def result
+      max_possible_complexity
+    end
+
+    # NOTE: I can't figure out how to trigger this with a test.
```
Comment (toyhammered) [inline app/graphql/analysis/max_query_complexity.rb]: @NuckChorris This is the issue I am trying to solve.

**Review by NuckChorris — APPROVED** [review]
Looks good! I see the analyzer API is less than ideal but you seem to have made it work!👌

**Comment by wopian** [comment]
LGTM USER-4195 🥇

**Comment by toyhammered** [comment]
> LGTM [USER-4195](https://kitsu.io/users/4195) 🥇

you forgot the s. I fixed it.

**Pull request merged**